src/Raven.Server/Documents/Indexes/MapReduce/AutoMapReduceIndex.cs

```diff
@@ -221,10 +210,20 @@ private unsafe void PutMappedResult(BlittableJsonReaderObject mappedResult, Lazy
             long storageId;
             if (existingEntry == null)
                 storageId = mapEntriesTable.Insert(tvb);
+            else if (existingEntry.ReduceKeyHash == reduceKeyHash)
+            {
+                // no need to update record in table since we have the same entry already stored
+                storageId = existingEntry.StorageId;
+            }
             else
             {
-                // no need to update since we have the same entry already stored
+                var previousState = GetReduceKeyState(existingEntry.ReduceKeyHash, indexContext, create: false);
+
                 storageId = existingEntry.StorageId;
+
+                previousState.Tree.Delete(new Slice((byte*)&storageId, sizeof(long)));
+
+                storageId = mapEntriesTable.Update(existingEntry.StorageId, tvb);
             }
 
             var pos = state.Tree.DirectAdd(new Slice((byte*)&storageId, sizeof(long)), mappedResult.Size);
```

Comment:
I don't follow what is going on. Why aren't you storing the value `mappedResult` directly in the `mapEntriesTable` ?
Why have _another_ level of indirection